File: runme/mcgen/local/datan/exasrc/e2sm.f
      PROGRAM E2SM
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      DIMENSION HIST(1000),SAMPLE(1)
      CHARACTER*75 TX,TY,CAPT
C identify program to user
      WRITE(*,*)' Program E1SM demonstrates the use of'
      WRITE(*,*)' subroutines SMHSIN, SMHSFL, SMHSGR'
      WRITE(*,*)' on a sample taken from the '
      WRITE(*,*)' standardized normal distribution'
      WRITE(*,*)' '
C ask for input
      WRITE(*,*)' Enter number N of elements in sample (>0)'
      WRITE(*,*)' >'
      READ(*,*)N
      WRITE(*,*)' Enter lower edge X0 of histogram'
      WRITE(*,*)' >'
      READ(*,*)X0
      WRITE(*,*)' Enter bin width DELX of histogram'
      WRITE(*,*)' >'
      READ(*,*)DELX
      WRITE(*,*)' Enter number NX of histogram bins (<1001)'
      WRITE(*,*)' >'
      READ(*,*)NX
C define texts
      TX='x'
      LTX=1
      TY='N(x)'
      LTY=4
      CAPT='N = ******'
      LCAPT=10
      WRITE(CAPT(5:10),'(I6)')N
C initialize histogram
      CALL SMHSIN(HIST,X0,DELX,NX)
C fill histogram
      DO 10 I=1,N
        CALL RNSTNR(SAMPLE,1)
        CALL SMHSFL(HIST,X0,DELX,NX,SAMPLE(1),DBLE(1.))
10    CONTINUE
C graphical output
      CALL GRNBWS()
      WRITE(*,*)' Please, enter number of workstation:'
      WRITE(*,*)' >'
      READ(*,*)NWS
      CALL SMHSGR(HIST,X0,DELX,NX,TX,LTX,TY,LTY,CAPT,LCAPT,NWS)
      END

File: runme/mcgen/local/datan/exasrc/e2sm.c
#include <stdio.h>
#include <string.h>
#include "datsrc.h"
#include "grsrc.h"

/* Table of constant values */

static integer c__1 = 1;
static double c_d1 = 1.;

main(void)
{
    /* Local variables */
    static integer ltx, lty, nws, i, n, lcapt, nx;
    static double delx, hist[1000], x0, sample[1];
    static char capt[75], tx[75], ty[75];

/* identify program to user */
    printf("%s\n", "Program E1SM demonstrates the use of");
    printf("%s\n", "subroutines SMHSIN, SMHSFL, SMHSGR");
    printf("%s\n", "on a sample taken from the");
    printf("%s\n\n", "standardized normal distribution.");
/* ask for input */
    printf("%s\n", "Enter number N of elements in sample (>0):");
    printf("%s", "> ");
    scanf("%ld", &n);
    printf("%s\n", "Enter lower edge X0 of histogram:");
    printf("%s", "> ");
    scanf("%lf", &x0);
    printf("%s\n", "Enter bin width DELX of histogram:");
    printf("%s", "> ");
    scanf("%lf", &delx);
    printf("%s\n", "Enter number NX of histogram bins (<1001):");
    printf("%s", "> ");
    scanf("%ld", &nx);
/* define texts */
    memset(tx, ' ', 75);
    memset(ty, ' ', 75);
    memset(capt, ' ', 75);
    strncpy(tx, "x", 1);
    ltx = 1;
    strncpy(ty, "N(x)", 4);
    lty = 4;
    strncpy(capt, "N = ******", 10);
    sprintf(&capt[4], "%6ld", n);
    lcapt = strlen(capt);
/* initialize histogram */
    smhsin_(hist, &x0, &delx, &nx);
/* fill histogram */
    for (i = 1; i <= n; ++i) {
        rnstnr_(sample, &c__1);
        smhsfl_(hist, &x0, &delx, &nx, sample, &c_d1);
    }
/* graphical output */
    grnbws_();
    printf("%s\n", "Please, enter number of workstation:");
    printf("%s", "> ");
    scanf("%ld", &nws);
    (void)getchar();
    smhsgr_(hist, &x0, &delx, &nx, tx, &ltx, ty, &lty, capt, &lcapt, &nws);
    return 0;
} /* main */

